src/platform/ESP32/ESP32BasicPacketFilters.h

```diff
@@ -0,0 +1,209 @@
+     *       count.
+     *
+     * @param maxAllowedQueuedPackets - number of packets currently pending allowed.
+     */
+    void SetMaxQueuedPacketsLimit(int maxAllowedQueuedPackets) { mMaxAllowedQueuedPackets.store(maxAllowedQueuedPackets); }
+
+    /**
+     * @return the total number of packets dropped so far by the filter
+     */
+    size_t GetNumDroppedPackets() const { return mNumDroppedPackets.load(); }
+
+    /**
+     * @brief Reset the counter of dropped packets.
+     */
+    void ClearNumDroppedPackets() { mNumDroppedPackets.store(0); }
+
+    /**
+     * @brief Method called when a packet is dropped due to high watermark getting reached, based on predicate.
+     *
+     * Subclasses may use this to implement additional behavior or diagnostics.
+     *
+     * This is called once for every dropped packet. If there is no filter predicate, this is not called.
+     *
+     * @param endpoint - pointer to endpoint instance (platform-dependent, which is why it's void)
+     * @param pktInfo - info about source/dest of packet
+     * @param pktPayload - payload content of packet
+     */
+    virtual void OnDropped(const void * endpoint, const chip::Inet::IPPacketInfo & pktInfo,
+                           const chip::System::PacketBufferHandle & pktPayload)
+    {}
+
+    /**
+     * @brief Method called whenever queue of accumulated packets is now empty, based on predicate.
+     *
+     * Subclasses may use this to implement additional behavior or diagnostics.
+     *
+     * This is possibly called repeatedly in a row, if the queue actually never gets above one.
+     *
+     * This is only called for packets that had matched the filtering rule, where they had
+     * been explicitly allowed in the past. If there is no filter predicate, this is not called.
+tcarmelveilleux marked this conversation as resolved.
+     *
+     * @param endpoint - pointer to endpoint instance (platform-dependent, which is why it's void)
+     * @param pktInfo - info about source/dest of packet
+     * @param pktPayload - payload content of packet
+     */
+    virtual void OnLastMatchDequeued(const void * endpoint, const chip::Inet::IPPacketInfo & pktInfo,
+                                     const chip::System::PacketBufferHandle & pktPayload)
+    {}
+
+    /**
+     * @brief Implementation of filtering before queueing that applies the predicate.
+     *
+     * See base class for arguments
+     */
+    FilterOutcome FilterBeforeEnqueue(const void * endpoint, const chip::Inet::IPPacketInfo & pktInfo,
+                                      const chip::System::PacketBufferHandle & pktPayload) override
+    {
+        // WARNING: This is likely called in a different context than `FilterAfterDequeue`. We use an atomic for the counter.
+        // Non-matching is never accounted, always allowed. Lack of predicate is equivalent to non-matching.
+        IPAddress::FromString("fe80::0000:1111:2222:3333", fakeDest);
```

Comment:
why are you parsing into these every time? Why are fakeDest and fakeMdnsDest global?

I would rather expect an init method to just parse once.

FromString returns a bool if the parsing is ok or not. Init should probably log if parsing failed.
the name `fake*` should explain what they mean and why we drop them

Reply:
Doing this in init function now,
Theses are the suspected packets which are causing overflow the Packet buffer Pool getting empty
We still need your thoughts @andy31415 @tcarmelveilleux on whether it can break anything in the code?